packages/astro/src/vite-plugin-build-html/index.ts

```diff
@@ -253,17 +259,46 @@ export function rollupPluginAstroBuildHTML(options: PluginOptions): VitePlugin {
         assetIdMap.set(assetPath, referenceId);
       }
 
+      // Sort CSS in order of appearance in HTML
+      // This is needed because bundling is an async process where the chunks may
+      // be processed in any order. So we have to do a little bit of work to determine
+      // chunk order based off of HTML scanning in options() earlier
+      const sortedCSSChunks = [...pureCSSChunks];
+      sortedCSSChunks.sort((a, b) => {
```

Comment:
Next, we have to do 2 sorts.

The first re-sorts the chunks in the HTML order derived above. This is basically the “global ordering.”